src/app/state/auth/auth.effects.ts

```diff
@@ -186,39 +169,35 @@ export class AuthEffects {
     { dispatch: false }
   );
 
-  reinitializeLoggedUser$ = createEffect(() =>
+  reinitializeUser$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(
-        AuthActions.reinitializeAuth,
-        AuthActions.reinitializeAuthForEnrol
-      ),
+      ofType(AuthActions.setMetamaskLoginOptions),
       filter(this.loginService.isSessionActive),
-      concatLatestFrom(() =>
-        this.store.select(AuthSelectors.isMetamaskDisabled)
-      ),
-      filter(([, isMetamaskDisabled]) => !isMetamaskDisabled),
-      concatLatestFrom(() => this.store.select(AuthSelectors.isUserLoggedIn)),
-      filter(([, isLoggedIn]) => !isLoggedIn),
-      tap(() => this.loadingService.show()),
-      switchMap(() =>
-        this.loginService
-          .login({ providerType: this.loginService.getSession().providerType })
-          .pipe(
-            map(({ success, accountInfo }) => {
-              if (success) {
-                return AuthActions.loginSuccess({ accountInfo });
-              }
-              return AuthActions.loginFailure();
-            }),
-            catchError((err) => {
-              console.log(err);
-              return of(AuthActions.loginFailure());
-            }),
-            finalize(() => {
-              this.loadingService.hide();
-            })
-          )
-      )
+      this.isLoggedIn(),
+      map(() => {
+        if (
+          this.loginService.getSession().providerType === ProviderType.MetaMask
+        ) {
+          return AuthActions.reinitializeAuthWithMetamask();
+        }
+
+        return AuthActions.reinitializeAuth();
+      })
+    )
+  );
+
+  reinitializeLoggedUserWithMetamask$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(AuthActions.reinitializeAuthWithMetamask),
+      this.isMetamaskDisabled(),
```

Comment:
Is this line the key difference between `reinitializeLoggedUserWithMetamask` and `reinitializeLoggedUser`? Where in the first one, additional information about whether or not metamask is disabled is required?

Reply:
Yes, isMetamaskDisabled means it has wrong chainId. I might rename it into for example `hasMetamaskCorrectChainId`.
`reinitializeLoggedUser` action is called when application tries reinitializing with different wallet than metamask (for example wallet connect).